### ISSUE TITLE
First copy() in main window fails

### ISSUE THREAD

**Issue by beefeater7** [hluk/CopyQ#1437]
**Describe the bug**
First encountered in [this thread](https://github.com/hluk/CopyQ/issues/1435):
>Copy in application once doesn't work, but spamming the command allows items to pass through

[This commit](https://github.com/hluk/CopyQ/commit/6ca729a69da5d357470e592d1a1d5b745efcac65) didn't fix the issue. It may be related to a [sister problem](https://github.com/hluk/CopyQ/issues/1438), but I don't know. Main window behaves different from Notepad: Always error on first copy, timing is irrelevant.

**To Reproduce**
```
[Command]
Name=Copy Repeater
Command="
    copyq:
    disable()
    
    function popupper() {
        try { 
            copy()
            popup(clipboard())
        } catch(e) { popup(e) }
    }
    
    for (i=0; i<10; i++) {
        popupper()
    }
    
    enable()"
IsGlobalShortcut=true
Icon=\xf15b
GlobalShortcut=shift+f4
```
1. Open main window
2. Select item(s)
3. Use command
4. Result
    ![image](https://user-images.githubusercontent.com/10468970/87259599-be0a8180-c4ac-11ea-9d92-da53f6ccd3c3.png)
    1 error followed by 9 popups of item contents.

**Expected behavior**
10 popups of item contents. The first is obviously most important.

**Version, OS and Environment**
```
CopyQ Clipboard Manager v3.12.0
Qt: 5.13.2
Compiler: GCC
Arch: x86_64-little_endian-llp64
OS: Windows 10 (10.0)
```

**Comment by beefeater7** [comment]
Fun little workaround:
```
[Command]
Name=Copy Repeater
Command="
    copyq:
    disable()
    
    function popupper() {
        try { 
            copy()
            popup(clipboard())
        } catch(e) { popupper() }
    }
    
    for (i=0; i<10; i++) {
        popupper()
    }
    
    enable()"
IsGlobalShortcut=true
Icon=\xf15b
GlobalShortcut=shift+f4
```
Yet, first copy takes forever to fall through, until error gets thrown. @pbodnar

**Comment by pbodnar** [comment]
@beefeater7, yes, I can confirm that. As in #1438, this seems to be a new bug introduced in v3.12.0, so I guess there will be a common cause.

> Yet, first copy takes forever to fall through, until error gets thrown. @pbodnar

Yes, this seems to be intentional - CopyQ waits for a new clipboard content for a while, as introduced by [Retry checking if copy() succeeded or throw an error](https://github.com/hluk/CopyQ/commit/a28aa3f2cb6d2f71a165ea224a6a69a244151f7e).

**Comment by beefeater7** [comment]
I can confirm also that [auto-repeat](https://github.com/hluk/CopyQ/issues/1438#issuecomment-658056940) setting has no effect on this issue. 

I found another edge case where the first copy fails, in global context actually. I'll see if I can isolate it

**Comment by hluk** [comment]
I believe this is same as #1438.

**Comment by beefeater7** [comment]
No, they are actually not. The command used is the same, but not the manner it is used or the problem it triggers.

Do we really have to use a wrapper function to make `copy()` work in the main window? Because _you_ don't see the issue? The first `copy()` event clearly misfires, I [catch the error and try again](https://github.com/hluk/CopyQ/issues/1437#issuecomment-657306638), voila it works. (After several seconds of waiting.) I'm sure it's not unsolvable.

If I can't [access `input()` from a Global shortcut **in the main window**](https://github.com/hluk/CopyQ/issues/1435), and `copy()` once _isn't working_, how do I solve it? I can use my little workaround, make it work for me, yada yada. Or I can make an effort and report the issue. After that it's out of my hands, I don't know C++. 

But you are dismissing bugs because you haven't encountered them. I put the effort to make them **reproducible**, you don't have to _believe_ anything, just try it for yourself. One error, nine successful copies. Have you tried it?

**Comment by hluk** [comment]
OK, sorry I did not read the command and assumed it has something to with holding the shortcut for a long time.

**Comment by hluk** [comment]
Unfortunately, I cannot test this on Windows (my VM died some time ago ... again) and on Linux it works well (it's a different code but same idea with waiting on the key to be released).

I assume the error pops up if you remove the loop. Right?

Does the error appear after you release the Shift modifier?

**Comment by beefeater7** [comment]
No worries, thanks for reopening. Yes, that error is the norm for "first time" or one-time copy in main window. I tried without the loop. 

The second copy and so on in a loop will guaranteed succeed afaict, unless holding down keys, but 1st fails regardless. It's not constant but close to it.

**Comment by Chealer** [comment]
@beefeater7: I cannot reproduce with CopyQ 13.0 (on Windows 11 25H2). Can you?
I get the expected behavior.

**Comment by beefeater7** [comment]
@Chealer, I tried it with my current machine.
```
CopyQ Clipboard Manager 13.0.0
Qt: 6.8.3
KNotifications: 6.12.0
Compiler: MSVC
Arch: x86_64-little_endian-llp64
OS: Windows 10 Version 22H2
```
I ran into no issues. The [version preceding 3.12](https://github.com/hluk/CopyQ/releases/download/v3.11.1/copyq-v3.11.1.zip) also had no issue. 

I checked [3.12](https://github.com/hluk/CopyQ/releases/download/v3.12.0/copyq-v3.12.0.zip) again and did reproduce it. The same issue remained in [3.13](https://github.com/hluk/CopyQ/releases/download/v3.13.0/copyq-v3.13.0.zip). It was gone in [4.0.0](https://github.com/hluk/CopyQ/releases/download/v4.0.0/copyq-v4.0.0.zip) (maybe the [release page](https://github.com/hluk/CopyQ/releases?page=3) has clues why)

This was easy enough to test and localize since all portable and installer releases use the same appdata.
